lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -92,50 +93,67 @@ public System.Threading.Tasks.Task RunTask()
                 return _userTask.Call(null);
-            }).ContinueWith(runTask =>
+            }).ContinueWith((System.Threading.Tasks.Task<byte[]> runTask) =>
+            {
+                try
                 {
-                    try
+                    // Task failed.
+                    if (runTask.IsFaulted)
                     {
-                        // Task failed.
-                        if (runTask.IsFaulted)
-                        {
-                            Logger.Log(Level.Warning,
-                                string.Format(CultureInfo.InvariantCulture, "Task failed caused by exception [{0}]", runTask.Exception));
-                            _currentStatus.SetException(runTask.Exception);
-                            return;
-                        }
+                        OnTaskFailure(runTask);
+                        return;
+                    }
 
-                        if (runTask.IsCanceled)
-                        {
-                            Logger.Log(Level.Warning,
-                                string.Format(CultureInfo.InvariantCulture, "Task failed caused by task cancellation"));
-                            return;
-                        }
+                    if (runTask.IsCanceled)
+                    {
+                        Logger.Log(Level.Error,
+                            string.Format(CultureInfo.InvariantCulture, "Task failed caused by System.Threading.Task cancellation"));
+                        OnTaskFailure(runTask);
+                        return;
+                    }
 
-                        // Task completed.
-                        var result = runTask.Result;
-                        Logger.Log(Level.Info, "Task Call Finished");
-                        _currentStatus.SetResult(result);
-                        if (result != null && result.Length > 0)
-                        {
-                            Logger.Log(Level.Info, "Task running result:\r\n" + System.Text.Encoding.Default.GetString(result));
-                        }
+                    // Task completed.
+                    var result = runTask.Result;
+                    Logger.Log(Level.Info, "Task Call Finished");
+                    _currentStatus.SetResult(result);
+                    if (result != null && result.Length > 0)
+                    {
+                        Logger.Log(Level.Info, "Task running result:\r\n" + System.Text.Encoding.Default.GetString(result));
                     }
-                    finally
+                }
+                finally
+                {
+                    if (_userTask != null)
                     {
-                        if (_userTask != null)
-                        {
-                            _userTask.Dispose();
-                        }
-
-                        runTask.Dispose();
+                        _userTask.Dispose();
                     }
-                });
+
+                    runTask.Dispose();
```

Comment:
No, `runTask` is implemented by the user. `_userTask` is of type `System.Threading.Task`, which is created by us to invoke the `Call` function in `runTask`.